data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpService.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder.server;
+
+import com.amazon.dataprepper.model.event.DefaultEventMetadata;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.annotation.Post;
+import org.opensearch.dataprepper.peerforwarder.model.WireEvent;
+import org.opensearch.dataprepper.peerforwarder.model.WireEvents;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PeerForwarderHttpService {
+    private static final Logger LOG = LoggerFactory.getLogger(PeerForwarderHttpService.class);
+
+    private final RequestExceptionHandler requestExceptionHandler;
+    private final ObjectMapper objectMapper;
+
+    public PeerForwarderHttpService(final RequestExceptionHandler requestExceptionHandler, final ObjectMapper objectMapper) {
+        this.requestExceptionHandler = requestExceptionHandler;
+        this.objectMapper = objectMapper;
+    }
+
+    @Post
```

Comment:
Does this need a path? `/event/forward`

Reply:
I'm not sure if it's required here since we provide this while configuring annotated service in `ServerBuilder`.
https://github.com/opensearch-project/data-prepper/blob/1a2b10532cf0e8e6dc9454c0313cda36d10faa1e/data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpServerProvider.java#L73

Reply:
Thanks for pointing that out. That should cover it.